_posts/2016-03-12-o-que-sao-expressoes-regulares.md

```diff
@@ -0,0 +1,100 @@
+| `{n,m}` | Busca por um mínimo de `n` e um máximo de `m` ocorrências |
+| `{n,}` | Busca por no mínimo `n` ocorrências |
+| `{,n}` | Busca por no máximo `n` ocorrências |
+| `?` | Busca por 1 (uma) ou nenhuma ocorrência |
+| `+` | Busca por 1 (uma) ou mais ocorrências |
+| `*` | Busca por 0 (zero) ou mais ocorrências |
+
+#### Âncoras
+Definem as posições de referência para o _match_ da expressão regular.
+
+| Metacaracter | Funcionalidade |
+|-----------------|---------------:|
+| `^` | Define a posição do início do texto ou de uma nova linha (em caso de textos multi-linhas) |
+| `\A` | Define a posição do início do texto |
+| `$` | Define a posição do fim do texto ou de uma linha (em caso de textos multi-linhas) |
+| `Z` | Define a posição do fim do texto |
+| `b` | Define a posição de borda, ou seja, logo antes do início de uma palavra ou logo após o fim da palavra |
+| `B` | Define a posição de *não* borda |
+
+#### Agrupamento
+Definem grupos
+
+| Metacaracter | Funcionalidade |
+|-----------------|---------------:|
+| `(...)` | Define um grupo para aplicação de um quantificador |
+| `...|...` | Define uma alternativa para a aplicação da expressão. A expressão deve dar _match_ à direita ou à esquerda |
+| `\n` | Recupera um _match_ no grupo _n_|
+
+Viu como as expressões são bem simples e, na maioria dos casos, os caracteres até fazem algum sentido?
+
+## Qual as vantagens das expressões regulares
+A grande vantagem de usar expressões regulares é poupar tempo. Elas são uma maneira simples de encontrar padrões. Você poderia escrever um `if` gigantesco e varrear cada caracter de uma _string_ para validar o formato de um e-mail, por exemplo, ou pode usar um simples `[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}`.
+
+![Regex](https://imgs.xkcd.com/comics/regular_expressions.png)
+
+## Onde usar expressões regulares
+As aplicações regulares no dia a dia do desenvolvedor são muitas. Você pode aplicar expressões regulares desde o console de debug (sim, o Xcode te dá essa opção) até nas validações de formulários. Um caso bem comum de aplicação de expressões regulares é na validação de e-mails. Além de e-mails, inúmeros campos podem ser validados com expressões regulares: telefones, CEPs, senhas, nomes, endereços, número de cartões de crédito etc.
+
+## Como usar expressões regulares no iOS
+Usar expressão regular no iOS é muito simples. Existe uma classe (`NSRegularExpression`) que pede no seu construtor a expressão e as opções que serão aplicadas no momento do _match_. Assim, caso você queira uma função para validar um campo CEP, por exemplo, pode fazer:
+
+~~~ swift
+func validateZipCode(zipCode: String) -> Bool {
+	let regex = try! NSRegularExpression(pattern: "^[0-9]{2}[0-9]{3}-[0-9]{3}$", options: [.CaseInsensitive])
+
+	let regexResult = regex.firstMatchInString(zipCode, options:[], range: NSMakeRange(0, zipCode.characters.count)) != nil
```

Comment:
"Note also that NSRange() counts the length of NSString, i.e. the number of UTF-16 code points, so  characters.count should be utf16.count, otherwise it might crash e.g. with Emoji characters."
Não estou muito familiar com Regex no Swift, mas fez sentido, talvez se aplique ao snippet acima.
http://stackoverflow.com/a/30845969/338514